src/main/java/com/google/gcp_variant_transforms/library/VcfParserImpl.java

```diff
@@ -5,17 +5,40 @@
 import com.google.common.collect.ImmutableList;
 import com.google.inject.Singleton;
 import htsjdk.variant.vcf.VCFCodec;
+import htsjdk.variant.vcf.VCFHeader;
+import htsjdk.variant.vcf.VCFHeaderLine;
+import java.util.HashSet;
+import java.util.Set;
+
 
 /** Implementation of {@link VcfParser} service. */
 @Singleton
 public class VcfParserImpl implements VcfParser {
 
   private static final long serialVersionUID = 6787944179556195968L;
+  private VCFHeader vcfHeader;
 
   @Override
   public VCFCodec generateCodecFromHeaderLines(ImmutableList<String> headerLines) {
     VCFCodec vcfCodec = new VCFCodec();
     vcfCodec.readActualHeader(new HeaderIterator(headerLines));
     return vcfCodec;
   }
+
+  public void generateVcfHeaderFromHeaderLines(ImmutableList<String> headerLines){

```

Comment:
I think the VCFHeader can be directly generated by VCFCodec.gerHeader(). VCFCodec reading actual header has already iterated all headerlines

Reply:
Do you think it makes sense to include vcfCodec as a private variable in the class so we can create it once and then access it repeatedly?

Reply:
I think so. Or can we just set this.vcfHeader in the generateCodecFromHeaderLines method?

Reply:
I added both a private codec and private header variable. Let me know if it makes more sense to leave out the private codec

Reply:
Unfortunately storing private variable `VcfCodec` won't work for other cases, other than reading headers - it won't be accessed from the worker code, because it's not serializable. Instead we have to generate it on each variant once (look how we do it in `ConvertLintToVariantFn()`). Therefore it's better to just create a `vcfCodec` and discard it for now.